### ISSUE TITLE
Table can be created with quicklz compression - but not used when data is inserted.

### ISSUE THREAD

**Issue by KAdamsInCO** [greenplum-db/gpdb#5107]
### Greenplum version or build

OSS GPDB 5.8.1
ORCA 2.56.3

### OS version and uname -a
[root@localhost ~]# uname -a
Linux mdw 3.10.0-862.3.2.el7.x86_64 #1 SMP Mon May 21 23:36:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
[root@localhost ~]# cat /etc/redhat-release 
CentOS Linux release 7.5.1804 (Core)

### Expected behavior

Do not allow tables in the OSS version to be created if compresstype is set to quicklz, as quicklz is not available in OSS Greenplum.

Would it be possible to prevent tables from being created with quicklz compression, instead of allowing it to only be told later on during a data insert that quicklz is not supported in the OSS version?

### Actual behavior

Tables / partitions can be created in OSS GPDB 5.8.1 with compresstype set to quicklz.

### Step to reproduce the behavior

gpadmin=# create table test (col1 text) WITH (appendonly=true, orientation=row, compresstype=quicklz) distributed randomly;
CREATE TABLE

gpadmin=# \d test
Append-Only Table "public.test"
 Column | Type | Modifiers 
--------+------+-----------
 col1   | text | 
Compression Type: quicklz
Compression Level: 1
Block Size: 32768
Checksum: t
Distributed randomly

gpadmin=# insert into test values ('A');
ERROR:  quicklz compression not supported (quicklz_compression.c:28)  (seg0 10.0.1.200:40000 pid=73718) (cdbdisp.c:254)
gpadmin=# 

Thanks.

**Comment by Eulerizeit** [comment]
Thanks for reporting. This is a known issue that we will be addressing shortly as we refactor our build system in the, hopefully, near future.

**Comment by doty-pivotal** [comment]
@jchampio and @kyeap-pivotal - looks like @KAdamsInCO already filed this bug.

To add on to this issue - we encountered this issue when creating a table with `compresstype=zstd`

TLDR; `CREATE TABLE` succeeds even if compression type is not supported

### Repro'ing this bug on the workstation

Version of GPDB: 6.x (master). 

(credit @kyeap-pivotal):

1. clone the latest copy of master.
2. `./configure --with-perl --with-python --with-libxml --with-gssapi --prefix=/usr/local/gpdb6 --disable-orca`
3. `make install`
4. `make -C gpAux/gpdemo`
5. `source /usr/local/gpdb6/greenplum_path.sh`
6. `source ~/workspace/gpdb6/gpAux/gpdemo/gpdemo-env.sh`
7. `psql postgres`
8. `create table test (a int) with(appendonly=true, compresstype=zstd, orientation=column);`
9. `select * from test`

at this point an error will pop up:

```
ERROR:  Zstandard library is not supported by this build  (seg0 slice1 10.254.0.58:25432 pid=97816)
HINT:  Compile with --with-zstd to use Zstandard compression.
```

I would have expected that if table `test` was unsupported, then the table should not have been created.

### Context:

If this bug persists, it is possible (with `regression` being an example) to build a DB that can't be dumped.

We encountered this issue when trying to run `pg_dumpall` on the a database after running `make installcheck-world`.  The database at the end of this process has a tables that were created,

```
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.858024 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd10,seg-1,,dx7321,,sx1,"LOG","00000","statement: CREATE TABLE zstdtest_1 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=1);",,,,,,"CREATE TABLE zstdtest_1 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=1);",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.878940 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd11,seg-1,,dx7326,,sx1,"LOG","00000","statement: CREATE TABLE zstdtest_10 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=10);",,,,,,"CREATE TABLE zstdtest_10 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=10);",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.903211 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd12,seg-1,,dx7336,,sx1,"LOG","00000","statement: CREATE TABLE zstdtest_19 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=19);",,,,,,"CREATE TABLE zstdtest_19 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=19);",0,,"postgres.c",1596,
```

all the subsequent operations on that table failed (note the `... An exception was encountered during the execution of statement ...`:

```
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.858024 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd10,seg-1,,dx7321,,sx1,"LOG","00000","statement: CREATE TABLE zstdtest_1 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=1);",,,,,,"CREATE TABLE zstdtest_1 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=1);",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.878940 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd11,seg-1,,dx7326,,sx1,"LOG","00000","statement: CREATE TABLE zstdtest_10 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=10);",,,,,,"CREATE TABLE zstdtest_10 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=10);",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.903211 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd12,seg-1,,dx7336,,sx1,"LOG","00000","statement: CREATE TABLE zstdtest_19 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=19);",,,,,,"CREATE TABLE zstdtest_19 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=19);",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.932268 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd13,seg-1,,dx7345,,sx1,"LOG","00000","statement: INSERT INTO zstdtest_1 SELECT g, 'foo' || g FROM generate_series(1, 10000) g;",,,,,,"INSERT INTO zstdtest_1 SELECT g, 'foo' || g FROM generate_series(1, 10000) g;",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.965566 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,8515,con111,cmd14,seg-1,,dx7345,x8515,sx1,"ERROR","0A000","Zstandard library is not supported by this build  (seg0 10.254.0.58:25432 pid=22002)",,"Compile with --with-zstd to use Zstandard compression.",,,,"INSERT INTO zstdtest_1 SELECT g, 'foo' || g FROM generate_series(1, 10000) g;",0,,"zstd_compression.c",156,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.965640 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,8515,con111,cmd14,seg-1,,dx7345,x8515,sx1,"LOG","00000","An exception was encountered during the execution of statement: INSERT INTO zstdtest_1 SELECT g, 'foo' || g FROM generate_series(1, 10000) g;",,,,,,,0,,,,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.966824 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd15,seg-1,,dx7360,,sx1,"LOG","00000","statement: INSERT INTO zstdtest_1 SELECT g, 'bar' || g FROM generate_series(1, 10000) g;",,,,,,"INSERT INTO zstdtest_1 SELECT g, 'bar' || g FROM generate_series(1, 10000) g;",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.993381 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,8519,con111,cmd16,seg-1,,dx7360,x8519,sx1,"ERROR","0A000","Zstandard library is not supported by this build  (seg0 10.254.0.58:25432 pid=22002)",,"Compile with --with-zstd to use Zstandard compression.",,,,"INSERT INTO zstdtest_1 SELECT g, 'bar' || g FROM generate_series(1, 10000) g;",0,,"zstd_compression.c",156,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:51.993421 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,8519,con111,cmd16,seg-1,,dx7360,x8519,sx1,"LOG","00000","An exception was encountered during the execution of statement: INSERT INTO zstdtest_1 SELECT g, 'bar' || g FROM generate_series(1, 10000) g;",,,,,,,0,,,,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.005166 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd17,seg-1,,dx7370,,sx1,"LOG","00000","statement: SELECT * FROM zstdtest_1 ORDER BY id LIMIT 5;",,,,,,"SELECT * FROM zstdtest_1 ORDER BY id LIMIT 5;",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.020493 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd18,seg-1,,dx7370,,sx1,"ERROR","0A000","Zstandard library is not supported by this build  (seg0 slice1 10.254.0.58:25432 pid=22002)",,"Compile with --with-zstd to use Zstandard compression.",,,,"SELECT * FROM zstdtest_1 ORDER BY id LIMIT 5;",0,,"zstd_compression.c",156,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.020554 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd18,seg-1,,dx7370,,sx1,"LOG","00000","An exception was encountered during the execution of statement: SELECT * FROM zstdtest_1 ORDER BY id LIMIT 5;",,,,,,,0,,,,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.021668 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd19,seg-1,,dx7386,,sx1,"LOG","00000","statement: SELECT * FROM zstdtest_1 ORDER BY id DESC LIMIT 5;",,,,,,"SELECT * FROM zstdtest_1 ORDER BY id DESC LIMIT 5;",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.080235 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd20,seg-1,,dx7386,,sx1,"ERROR","0A000","Zstandard library is not supported by this build  (seg0 slice1 10.254.0.58:25432 pid=22002)",,"Compile with --with-zstd to use Zstandard compression.",,,,"SELECT * FROM zstdtest_1 ORDER BY id DESC LIMIT 5;",0,,"zstd_compression.c",156,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.080266 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd20,seg-1,,dx7386,,sx1,"LOG","00000","An exception was encountered during the execution of statement: SELECT * FROM zstdtest_1 ORDER BY id DESC LIMIT 5;",,,,,,,0,,,,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.081782 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd21,seg-1,,dx7412,,sx1,"LOG","00000","statement: INSERT INTO zstdtest_19 SELECT g, 'foo' || g FROM generate_series(1, 10000) g;",,,,,,"INSERT INTO zstdtest_19 SELECT g, 'foo' || g FROM generate_series(1, 10000) g;",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.109878 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,8544,con111,cmd22,seg-1,,dx7412,x8544,sx1,"ERROR","0A000","Zstandard library is not supported by this build  (seg1 10.254.0.58:25433 pid=22006)",,"Compile with --with-zstd to use Zstandard compression.",,,,"INSERT INTO zstdtest_19 SELECT g, 'foo' || g FROM generate_series(1, 10000) g;",0,,"zstd_compression.c",156,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.109927 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,8544,con111,cmd22,seg-1,,dx7412,x8544,sx1,"LOG","00000","An exception was encountered during the execution of statement: INSERT INTO zstdtest_19 SELECT g, 'foo' || g FROM generate_series(1, 10000) g;",,,,,,,0,,,,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.110103 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd23,seg-1,,dx7420,,sx1,"LOG","00000","statement: INSERT INTO zstdtest_19 SELECT g, 'bar' || g FROM generate_series(1, 10000) g;",,,,,,"INSERT INTO zstdtest_19 SELECT g, 'bar' || g FROM generate_series(1, 10000) g;",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.140272 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,8550,con111,cmd24,seg-1,,dx7420,x8550,sx1,"ERROR","0A000","Zstandard library is not supported by this build  (seg0 10.254.0.58:25432 pid=22002)",,"Compile with --with-zstd to use Zstandard compression.",,,,"INSERT INTO zstdtest_19 SELECT g, 'bar' || g FROM generate_series(1, 10000) g;",0,,"zstd_compression.c",156,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.140316 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,8550,con111,cmd24,seg-1,,dx7420,x8550,sx1,"LOG","00000","An exception was encountered during the execution of statement: INSERT INTO zstdtest_19 SELECT g, 'bar' || g FROM generate_series(1, 10000) g;",,,,,,,0,,,,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.141250 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd25,seg-1,,dx7430,,sx1,"LOG","00000","statement: SELECT * FROM zstdtest_19 ORDER BY id LIMIT 5;",,,,,,"SELECT * FROM zstdtest_19 ORDER BY id LIMIT 5;",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.173352 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd26,seg-1,,dx7430,,sx1,"ERROR","0A000","Zstandard library is not supported by this build  (seg0 slice1 10.254.0.58:25432 pid=22002)",,"Compile with --with-zstd to use Zstandard compression.",,,,"SELECT * FROM zstdtest_19 ORDER BY id LIMIT 5;",0,,"zstd_compression.c",156,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.173736 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd26,seg-1,,dx7430,,sx1,"LOG","00000","An exception was encountered during the execution of statement: SELECT * FROM zstdtest_19 ORDER BY id LIMIT 5;",,,,,,,0,,,,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.173952 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd27,seg-1,,dx7438,,sx1,"LOG","00000","statement: SELECT * FROM zstdtest_19 ORDER BY id DESC LIMIT 5;",,,,,,"SELECT * FROM zstdtest_19 ORDER BY id DESC LIMIT 5;",0,,"postgres.c",1596,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.204067 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd28,seg-1,,dx7438,,sx1,"ERROR","0A000","Zstandard library is not supported by this build  (seg0 slice1 10.254.0.58:25432 pid=22002)",,"Compile with --with-zstd to use Zstandard compression.",,,,"SELECT * FROM zstdtest_19 ORDER BY id DESC LIMIT 5;",0,,"zstd_compression.c",156,
gpdb-2018-07-25_165453.csv:2018-07-25 16:57:52.204102 UTC,"gpadmin","regression",p21958,th1776170784,"[local]",,2018-07-25 16:57:51 UTC,0,con111,cmd28,seg-1,,dx7438,,sx1,"LOG","00000","An exception was encountered during the execution of statement: SELECT * FROM zstdtest_19 ORDER BY id DESC LIMIT 5;",,,,,,,0,,,,
```

, and when it comes time to dump the database, `pg_dumpall` fails with 

```
pg_dump: SQL command failed
pg_dump: Error message from server: ERROR:  current transaction is aborted, commands ignored until end of transaction block  (seg0 10.254.0.58:25432 pid=100461)
pg_dump: The command was: COPY public.zstdtest_1 (id, t) TO stdout IGNORE EXTERNAL PARTITIONS;
pg_dumpall: pg_dump failed on database "regression", exiting
```

### Separate but related question: 

Why does the CI system report that 
```
compression_zstd         ... ok (0.60 sec)  (diff:0.16 sec)
```
, if the `src/test/regress/sql/compression_zstd.sql`  `INSERTS`'s and `SELECT`'s don't succeed?

**Comment by doty-pivotal** [comment]
@asimrp:

We looked at `5X_STABLE` and didn't see the `zstd` commit backported (724f9d27b160ec3f87a5ce95eecc605808be3ced went into master).

We can repro this on `5X_STABLE` for `compresstype=quicklz`  by configuring with the following options:

```
$ ./configure --with-perl --with-python --with-libxml --with-gssapi --prefix=/usr/local/gpdb5 --disable-orca

... same demo cluster dance as above ...

postgres=#  create table test (a int) with(appendonly=true, compresstype=quicklz, orientation=column) distributed randomly;
CREATE TABLE
postgres=# select * from test;
ERROR:  quicklz compression not supported (quicklz_compression.c:28)  (seg0 slice1 127.0.0.1:25432 pid=21484) (cdbdisp.c:254)
```

**Comment by jimmyyih** [comment]
> We looked at 5X_STABLE and didn't see the zstd commit backported (724f9d2 went into master).

Yes, `zstd` will not be backported to 5X_STABLE as it requires a catalog change.

> Why does the CI system report that [the zstd tests are passing]

There are two separate answer files.  One answer file contains the output if you have zstd compiled (compression_zstd.out) and the other is used if you do not have it compiled (compression_zstd_1.out).

> all the subsequent operations on that table failed (followed by log output)

That log output looks like the expected output of when you run the zstd regress test without zstd compiled.  It should have no affect on `pg_dumpall`.

> and when it comes time to dump the database (followed by pg_dumpall log output)

Interesting.  Here's a smaller repro:
```
postgres=# CREATE TABLE zstdtest_1 (id int4, t text) WITH (appendonly=true, compresstype=zstd, compresslevel=1);
CREATE TABLE
postgres=# COPY public.zstdtest_1 (id, t) TO stdout IGNORE EXTERNAL PARTITIONS;
postgres=# COPY public.zstdtest_1 (id, t) TO stdout IGNORE EXTERNAL PARTITIONS;
postgres=# BEGIN;
BEGIN
postgres=# COPY public.zstdtest_1 (id, t) TO stdout IGNORE EXTERNAL PARTITIONS;
postgres=# COPY public.zstdtest_1 (id, t) TO stdout IGNORE EXTERNAL PARTITIONS;
ERROR:  current transaction is aborted, commands ignored until end of transaction block  (seg0 127.0.0.1:25432 pid=9496)
```
I would have expected `COPY` to bubble up the error message `ERROR:  Zstandard library is not supported by this build` back to the user.
